### ISSUE TITLE
Make vCategory iterable

### ISSUE THREAD

**Issue by niccokunzmann** [collective/icalendar#610]
vCategory is used to list different categories of events. It would be nice to have an easy use of that.

```
from icalendar.prop import vCategory
c = vCategory.from_ical("APPOINTMENT,EDUCATION")
cats = list(c) # new feature
assert cats == ["APPOINTMENT", "EDUCATION"]
```

**Comment by devdanzin** [comment]
This seems like a good feature, but is the code example correct? Because from:

https://github.com/collective/icalendar/blob/39fef78ec0dc221e3238be6d1b73b198b696056f/src/icalendar/tests/prop/test_unit.py#L331-L339

It seems `vCategory.from_ical` doesn't return an instance of vCategory, but a list.

So should we work from:
```python
from icalendar.prop import vCategory
c = vCategory(vCategory.from_ical("APPOINTMENT,EDUCATION"))
cats = list(c) # new feature
assert cats == ["APPOINTMENT", "EDUCATION"]
```
instead?

**Comment by niccokunzmann** [comment]
Hi, a list sounds good to me. I think that the documentation should reflect it so that it is clear how to use it.

**Comment by niccokunzmann** [comment]
Fixed in #670.
